pkg/process/checks/system_info_darwin_test.go

```diff
@@ -36,3 +37,19 @@ func TestPatchCPUInfo(t *testing.T) {
 		assert.Equal(t, int32(2), c.Cores)
 	}
 }
+
+func TestOnlyCorePopulatedDarwin(t *testing.T) {
+	sysInfo, _ := CollectSystemInfo()
+	for _, cpuData := range sysInfo.Cpus {
+		// Checks if only the cores does not have the default value
+		assert.NotEqual(t, int32(0), cpuData.Cores)
```

Comment:
You should use [assert.Greater()](https://pkg.go.dev/github.com/stretchr/testify/assert#Greater) here. It shouldn't be negative, but this is expresses the intent more clearly.